src/dmd/libomf.d

```diff
@@ -167,33 +167,37 @@ final class LibOMF : Library
 
     /*****************************************************************************/
 
-    void addSymbol(OmfObjModule* om, const(char)* name, int pickAny = 0)
+    void addSymbol(OmfObjModule* om, const(char)[] name, int pickAny = 0)
     {
+        assert(name.length == strlen(name.ptr));
         static if (LOG)
         {
-            printf("LibOMF::addSymbol(%s, %s, %d)\n", om.name.ptr, name, pickAny);
+            printf("LibOMF::addSymbol(%.*s, %.*s, %d)\n",
+                cast(int)om.name.length, om.name.ptr,
+                cast(int)name.length, name.ptr, pickAny);
         }
-        const namelen = strlen(name);
-        StringValue* s = tab.insert(name, namelen, null);
-        if (!s)
+        if (auto s = tab.insert(name, null))
+        {
+            auto os = new OmfObjSymbol();
+            os.name = strdup(name.ptr);
```

Comment:
At this point, we are assured that name.length==strlen(name.ptr), but strdup(name.ptr) recalculates strlen(name.ptr). Also, strdup returns null when it fails, and no check is done. I know the check wasn't done in the original, either, but since you're redoing the code anyway that should be fixed.

Reply:
Check added.

Reply:
strup is still recalculating the length.